llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -552,6 +552,14 @@ Value *InstCombinerImpl::SimplifyDemandedUseBits(Value *V, APInt DemandedMask,
     if (DemandedFromOps.isSubsetOf(LHSKnown.Zero))
       return I->getOperand(1);
 
+    // (add X, C) --> (xor X, C) IFF C is equal to the top bit of the DemandMask
+    {
+      const APInt *C;
+      if (match(I->getOperand(1), m_APInt(C)) &&
+          C->isOneBitSet(DemandedMask.getActiveBits() - 1))
+        return Builder.CreateXor(I->getOperand(0), ConstantInt::get(VTy, *C));
```

Comment:
```
Instruction *Xor = BinaryOperator::CreateXor(I->getOperand(0), ConstantInt::get(VTy, *C));
return InsertNewInstWith(Xor, I->getIterator());
```
I guess it will fix the assertion failure.

Reply:
Ah, sorry for missing that. As this is in SimplifyDemanded we should explicitly set the IRBuilder InsertPoint.